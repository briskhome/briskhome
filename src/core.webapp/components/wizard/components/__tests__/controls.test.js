import React from 'react';
import toJson from 'enzyme-to-json';
import { mount, render } from 'enzyme';

import { WizardControls } from '../../';

const MockSlide = jest.fn(() => <div data-mock="MockSlide" />);

describe('core.webapp -> components -> wizard -> controls', () => {
  const prev = jest.fn();
  const next = jest.fn();

  beforeEach(() => {
    jest.restoreAllMocks();
  });
  it('renders', () => {
    expect(
      toJson(render(<WizardControls prev={prev} next={next} />)),
    ).toMatchSnapshot();
  });
  it('renders without "prev" button', () => {
    expect(toJson(render(<WizardControls next={next} />))).toMatchSnapshot();
  });
  it('renders without "next" button', () => {
    expect(toJson(render(<WizardControls prev={prev} />))).toMatchSnapshot();
  });
  it('does not render', () => {
    expect(toJson(render(<WizardControls />))).toMatchSnapshot();
  });
});
