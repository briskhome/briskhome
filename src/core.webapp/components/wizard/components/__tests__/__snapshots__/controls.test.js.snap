// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`core.webapp -> components -> wizard -> controls does not render 1`] = `null`;

exports[`core.webapp -> components -> wizard -> controls renders 1`] = `
<div
  class="briskhome-wizard__footer-controls briskhome-wizard__footer-controls_left briskhome-wizard__footer-controls_right briskhome-wizard__footer-controls_justify"
>
  <a
    class="briskhome-button briskhome-button_inline-block briskhome-wizard__button-prev"
    href="#"
    role="button"
    target="_self"
    type="button"
  >
    Go Back
  </a>
  <a
    class="briskhome-button briskhome-button_yellow briskhome-button_caps briskhome-button_inline-block briskhome-wizard__button-next"
    href="#"
    role="button"
    target="_self"
    type="button"
  >
    Next
  </a>
</div>
`;

exports[`core.webapp -> components -> wizard -> controls renders without "next" button 1`] = `
<div
  class="briskhome-wizard__footer-controls briskhome-wizard__footer-controls_left"
>
  <a
    class="briskhome-button briskhome-button_inline-block briskhome-wizard__button-prev"
    href="#"
    role="button"
    target="_self"
    type="button"
  >
    Go Back
  </a>
</div>
`;

exports[`core.webapp -> components -> wizard -> controls renders without "prev" button 1`] = `
<div
  class="briskhome-wizard__footer-controls briskhome-wizard__footer-controls_right"
>
  <a
    class="briskhome-button briskhome-button_yellow briskhome-button_caps briskhome-button_inline-block briskhome-wizard__button-next"
    href="#"
    role="button"
    target="_self"
    type="button"
  >
    Next
  </a>
</div>
`;
